gridstatus/ercot.py

```diff
@@ -576,6 +579,120 @@ def get_spp(
             )
         return Ercot._finalize_spp_df(df, settlement_point_field, locations)
 
+    def get_as_prices(
+        self,
+        date,
+        end=None,
+        verbose=False,
+    ):
+        """Get ancillary service clearing prices in hourly intervals in Day Ahead Market
+
+        Arguments:
+            date (datetime.date, str): date of delivery for AS services
+
+            end (datetime.date, str, optional): if declared, function will return
+                data as a range, from "date" to "end"
+
+            verbose (bool, optional): print verbose output. Defaults to False.
+
+        Returns:
+
+            pandas.DataFrame: A DataFrame with prices for "Non-Spinning Reserves", \
+                "Regulation Up", "Regulation Down", "Responsive Reserves".
+
+        Source:
+            https://www.ercot.com/mp/data-products/data-product-details?id=NP4-181-ER
+        """
+
+        if not end:
+            return self._get_as_prices(date)
+
+        start = utils._handle_date(date, tz=self.default_timezone)
+
+        # add one day since end is exclusive
+        end = utils._handle_date(end, tz=self.default_timezone) + pd.DateOffset(days=1)
+
+        # annual files, so get all years we'll need to iter over
+        years = {x for x in range(start.year, end.year + 1)}
```

Comment:
if you use the `@support_date_range("1Y")` decorator you shouldn't get any calls to this function that have more than one year